src/Microsoft.DotNet.Tools.Compiler.Native/IntermediateCompilation/Mac/MacRyuJitCompileStep.cs

```diff
@@ -0,0 +1,146 @@
+
+        private readonly string CompilerOutputExtension = "";
+
+        // TODO: debug/release support
+        private readonly string cflags = "-g -lstdc++ -Wno-invalid-offsetof -pthread -ldl -lm -liconv";
+
+        private readonly string[] libs = new string[]
+        {
+            "libbootstrapper.a",
+            "libRuntime.a",
+            "libSystem.Private.CoreLib.Native.a"
+        };
+
+        private readonly string[] appdeplibs = new string[]
+        {
+            "libSystem.Native.a"
+        };
+
+
+        private string CompilerArgStr { get; set; }
+        private NativeCompileSettings config;
+
+        public MacRyuJitCompileStep(NativeCompileSettings config)
+        {
+            this.config = config;
+            InitializeArgs(config);
+        }
+
+        public int Invoke()
+        {
+            Console.WriteLine("MacRyuJitCompileStep: Invoking JIt step");
+            var result = InvokeCompiler();
+            if (result != 0)
+            {
+                Reporter.Error.WriteLine("Compilation of intermediate files failed.");
+            }
+
+            return result;
+        }
+
+        public bool CheckPreReqs()
+        {
+            // TODO check for clang
+            return true;
+        }
+
+        private void InitializeArgs(NativeCompileSettings config)
+        {
+            var argsList = new List<string>();
+
+            // Flags
+            argsList.Add(cflags);
+            
+            // Add Stubs
+            argsList.Add("-I "+Path.Combine(config.AppDepSDKPath, "CPPSdk/osx.10.10"));
+            argsList.Add("-I "+Path.Combine(config.AppDepSDKPath, "CPPSdk"));
+            argsList.Add(Path.Combine(config.AppDepSDKPath, "CPPSdk/osx.10.10/osxstubs.cpp"));
+
+            // Input File
+            var inLibFile = DetermineInFile(config);
+            argsList.Add("-Xlinker "+inLibFile);
+
+            // Libs
+            foreach (var lib in libs)
+            {
+                var libPath = Path.Combine(config.IlcPath, lib);
+                argsList.Add("-Xlinker "+libPath);
+            }
+
+            // AppDep Libs
+            var baseAppDepLibPath = Path.Combine(config.AppDepSDKPath, "CPPSdk/osx.10.10", config.Architecture.ToString());
+            foreach (var lib in appdeplibs)
+            {
+                var appDepLibPath = Path.Combine(baseAppDepLibPath, lib);
+                argsList.Add("-Xlinker "+appDepLibPath);
+            }
+
+            // Output
+            var libOut = DetermineOutputFile(config);
+            argsList.Add($"-o \"{libOut}\"");
+
+            this.CompilerArgStr = string.Join(" ", argsList);
+        }
+
+        private int InvokeCompiler()
+        {
+            var result = Command.Create(CompilerName, CompilerArgStr)
+                .ForwardStdErr()
+                .ForwardStdOut()
+                .Execute();
+
+            // Needs System.Native.dylib PAL in output - ideally, this should come as app-dependency,
```

Comment:
Per @brthor's comment, @gkhanna79, please remove it for Linux as well while you are at it.